test/martha/martha_v3.test.js

```diff
@@ -281,28 +363,38 @@ test.serial('martha_v3 returns null for fields missing in drs and bond response'
     const response = mockResponse();
     await marthaV3(mockRequest({ body: { 'url': 'drs://abc/123' } }), response);
     const result = response.send.lastCall.args[0];
-    t.deepEqual(Object.assign({}, result), expectedObjWithMissingFields);
+    t.deepEqual({ ...result }, expectedObjWithMissingFields);
     t.is(response.statusCode, 200);
 });
 
+function generateUrl(testUrl) {
+    const parsedUrl = new URL(testUrl);
+    return determineDrsType(parsedUrl).urlGenerator(parsedUrl);
+}
 
 /**
  * determineDrsType(uri) -> drsUrl Scenario 1: data objects uri with non-dg host and path
  */
 test('determineDrsType should parse dos:// Data Object uri', (t) => {
```

Comment:
please update the description of the test to reflect the new logic
e.g.
```suggestion
test('generateUrl should return correct URL from dos:// Data Object URI', (t) => {
```

rinse, repeat for the rest of the `generateURL` tests